pkg/ipam-controller/allocator/allocator.go

```diff
@@ -200,186 +185,54 @@ type AllocationConfig struct {
 	Subnet           *net.IPNet
 	Gateway          net.IP
 	PerNodeBlockSize int
+	NodeSelector     *corev1.NodeSelector
 }
 
 func (pc *AllocationConfig) Equal(other *AllocationConfig) bool {
 	return reflect.DeepEqual(pc, other)
 }
 
-// New create and initialize new allocator
-func New() *Allocator {
-	return &Allocator{allocators: map[string]*poolAllocator{}}
-}
-
-type Allocator struct {
-	lock       sync.Mutex
-	allocators map[string]*poolAllocator
-	configured bool
-}
-
-// IsConfigured returns true if allocator is configured
-func (a *Allocator) IsConfigured() bool {
-	a.lock.Lock()
-	defer a.lock.Unlock()
-	return a.configured
-}
-
-// Configure update allocator configuration
-func (a *Allocator) Configure(ctx context.Context, configs []AllocationConfig) {
-	a.lock.Lock()
-	defer a.lock.Unlock()
-	a.configure(ctx, configs)
-}
-
-// ConfigureAndLoadAllocations configures allocator and load data from the node objects
-func (a *Allocator) ConfigureAndLoadAllocations(ctx context.Context, configs []AllocationConfig, nodes []corev1.Node) {
+// CreatePoolAllocatorFromIPPool creates a PoolAllocator and load data from the IPPool CR
```

Comment:
please add some info in docstring as for the role of nodes arg

Reply:
Done